### ISSUE TITLE
MPSC Channel test case times out.

### ISSUE THREAD

**Issue by YoshikiTakashima** [model-checking/kani#1286]
Tracking issue

_Originally posted by @danielsn in https://github.com/model-checking/kani/pull/1285#discussion_r899169729_

Running 
```bash'
./scripts/kani mpsc.rs
```
on the following code times out as of Kani 0.4.0 on M1 Mac, cargo 1.63.0-nightly (85e457e15 2022-06-07). Note that `kani::unwind(1)` fails to verify, and `kani::unwind(2)` or not putting an unwind limit leads to timeout.
```rust
// mpsc.rs
use std::sync::mpsc::*;

static mut CELL: i32 = 0;

struct DropSetCELLToOne {}

impl Drop for DropSetCELLToOne {
    fn drop(&mut self) {
        unsafe {
            CELL = 1;
        }
    }
}

#[kani::unwind(2)]
#[kani::proof]
fn main() {
    {
        let (send, recv) = channel::<DropSetCELLToOne>();
        send.send(DropSetCELLToOne {}).unwrap();
        let _to_drop: DropSetCELLToOne = recv.recv().unwrap();
    }
    assert_eq!(unsafe { CELL }, 1, "Drop should be called");
}
```

**Comment by tedinski** [comment]
Just to clarify: It times out _with_ the `unwind(2)`? Or it times out without it and seems weird to need it?

**Comment by celinval** [comment]
Interesting, with the fix for #2164, this test passes on Mac but fails on Linux due to #1781. :(
